cypress/integration/dunning/change_text_dunning_doc.js

```diff
@@ -0,0 +1,134 @@
+import { DunningCandidates } from '../../page_objects/dunning_candidates';
+import { applyFilters, selectNotFrequentFilterWidget, toggleNotFrequentFilters } from '../../support/functions';
+import { DunningType } from '../../support/utils/dunning_type';
+import { DunningDocuments } from '../../page_objects/dunning_documents';
+import { salesInvoices } from '../../page_objects/sales_invoices';
+
+describe('Create Dunning Documents', function() {
+  const date = humanReadableNow();
+
+  const dunningTypeName = `Dunning ${date}`;
+
+  const businessPartnerName = `Customer Dunning ${date}`;
+  const paymentTerm = 'immediately';
+
+  const salesInvoiceTargetDocumentType = 'Sales Invoice';
+  const productName = 'Convenience Salat 250g';
+  const originalQuantity = 200;
+
+  let siDocumentNumber;
+  let dunningTypeRecordId;
+  let siRecordId;
+  let dunningDocId;
+
+  it('Create dunning type', function() {
+    cy.fixture('settings/dunning_type.json').then(dunningType => {
+      Object.assign(new DunningType(), dunningType)
+        .setName(dunningTypeName)
+        .apply();
+    });
+    cy.getCurrentWindowRecordId().then(recordId => {
+      dunningTypeRecordId = recordId;
+    });
+  });
+
+  it('Create bpartner', function() {
+    cy.fixture('sales/simple_customer.json').then(customerJson => {
+      const bpartner = new BPartner({ ...customerJson, name: businessPartnerName })
+        .setCustomer(true)
+        .setDunning(dunningTypeName)
+        .setPaymentTerm(paymentTerm)
+        .setBank(undefined);
+
+      bpartner.apply();
+    });
+  });
+
+  it('Create sales invoice', function() {
+    cy.fixture('sales/sales_invoice.json').then(salesInvoiceJson => {
+      new SalesInvoice(businessPartnerName, salesInvoiceTargetDocumentType)
+        .addLine(new SalesInvoiceLine().setProduct(productName).setQuantity(originalQuantity))
+        .apply();
+    });
+    cy.completeDocument();
+  });
+  it('Sales Invoice is not paid', function() {
+    cy.getCheckboxValue('IsPaid').then(checkBoxValue => {
+      cy.log(`IsPaid = ${checkBoxValue}`);
+      assert.equal(checkBoxValue, false);
+    });
+  });
+
+  it('Save document no from invoice', function() {
+    cy.getStringFieldValue('DocumentNo').then(documentNumber => {
+      siDocumentNumber = documentNumber;
+    });
+  });
+  it('Save current record id of the invoice', function() {
+    cy.getCurrentWindowRecordId().then(function(record) {
+      siRecordId = record;
+    });
+  });
+
+  it('Create Dunning Candidates', function() {
+    DunningCandidates.visit();
+    cy.executeHeaderActionWithDialog('C_Dunning_Candidate_Create');
+    cy.setCheckBoxValue('IsFullUpdate', true, true);
+    cy.pressStartButton();
+    // cy.wait(8000);
+  });
+
+  it('Ensure there are exactly 2 Dunning Candidates - one for each dunning type level', function() {
+    DunningCandidates.visit();
+    filterBySalesInvoiceDocNumber(siDocumentNumber);
+    DunningCandidates.getRows().should('have.length', 2);
```

Comment:
alternative: `cy.expectNumberOfRows(2)` (just fyi)

Reply:
you're right, done.